### DISCUSSION TITLE
Multi runner environment in a workflow.

### DISCUSSION THREAD

**Discussion by vansh2552** [community/community#58104] [Actions] [answered]
### Select Topic Area

Question

### Body

 I was working with multiple runner environments and I wanted to share cache and artifacts between them. On doing some research I found out that each runner has its own environment where artifacts and cache are stored so they cant be shared across multiple runners. But I just gave it a shot and to my surprise I am able to share artifacts between different runners, but I dont know how is that happening. Does anyone have an explanation on how this is happening, and also is it possible for caching aswell since I'm not sure how to test that.

**Comment by airtower-luna** [comment, marked as answer]
> On doing some research I found out that each runner has its own environment where artifacts and cache are stored so they cant be shared across multiple runners.

If you mean artifacts uploaded with `actions/upload-artifact` and caches created with `actions/cache` (or the equivalent APIs), those are uploaded to servers provided by GitHub and can be downloaded by other authorized runners. That's actually what they're there for.

**Reply by vansh2552** [reply]
so if I upload cache using actions/cache, then any runner in the workflow can access it with the cache key, right?

**Reply by airtower-luna** [reply]
If the cache is available by the time it's accessed, yes. If you want a reliable transfer of data between jobs it's usually better to use artifacts, though. :slightly_smiling_face:

**Reply by vansh2552** [reply]
Thanks a lot.

**Reply by saalim-mushtaq** [reply]
@vansh2552 Can I use cache actions to share cache between tasks running simultaneously with a matrix strategy?
